ksql-common/src/main/java/io/confluent/ksql/schema/ksql/LogicalSchema.java

```diff
@@ -123,12 +171,35 @@ public ConnectSchema getSchema() {
    * before attempting to find a match again.
    *
    * @param fieldName the field name, where any alias is ignored.
+   * @return the field if found, else {@code Optiona.empty()}.
    */
   public Optional<Field> findField(final String fieldName) {
-    return schema.fields()
-        .stream()
-        .filter(f -> SchemaUtil.matchFieldName(f, fieldName))
-        .findFirst();
+    Optional<Field> found = findSchemaField(fieldName, implicitSchema);
```

Comment:
Are we using java 9 yet? If so:
```java
return findSchemaField(fieldName, implicitSchema)
   .or(findSchemaField(fieldName, keySchema)
   .or(findSchemaField(fieldName, valueSchema));
```
Looks like checkstyle complains about this though :/

Reply:
We're still on 8 I believe.